src/model_repository_manager.h

```diff
@@ -260,65 +267,119 @@ class ModelRepositoryManager {
   Status InitializeModelInfo(
       const std::string& name, const std::string& path,
       const std::vector<const InferenceParameter*>& params,
-      std::unique_ptr<ModelInfo>* info);
+      std::unique_ptr<ModelInfo>* info) const;
 
   /// Load models based on the dependency graph. The function will iteratively
   /// load models that all the models they depend on has been loaded, and unload
   /// models if their dependencies are no longer satisfied.
+  /// \param infos The model infos.
+  /// \param dependency_graph The dependency graph.
   /// \return The status of the model loads.
-  std::map<std::string, Status> LoadModelByDependency();
+  std::map<std::string, Status> LoadModelByDependency(
+      const ModelInfoMap& infos, const DependencyGraph& dependency_graph) const;
 
   /// Helper function to update the dependency graph based on the poll result
   /// \param added The names of the models added to the repository.
   /// \param deleted The names of the models removed from the repository.
   /// \param modified The names of the models remaining in the
   /// repository that have been changed.
+  /// \param model_infos The model infos.
+  /// \param updated_graph The dependency graph.
   /// \param deleted_dependents The names of dependent models to be removed
   /// from the repository.
   /// \return The error status.
   Status UpdateDependencyGraph(
       const std::set<std::string>& added, const std::set<std::string>& deleted,
-      const std::set<std::string>& modified,
-      std::set<std::string>* deleted_dependents = nullptr);
+      const std::set<std::string>& modified, const ModelInfoMap& model_infos,
+      DependencyGraph* updated_graph,
+      std::set<std::string>* deleted_dependents = nullptr) const;
+
+  /// Helper function to deep copy the dependency graph
+  /// \param new_graph The new graph to be cleared and copied into
+  void CopyDependencyGraph(DependencyGraph* new_graph) const;
+
+  /// Helper function to re-map dependency graph nodes pointers from old nodes
+  /// to new nodes.
+  /// The function search for matching pointers in the new_nodes first, if not
+  /// found, then the matching pointer must be in the ref_nodes.
```

Comment:
`ref_nodes` and `new_nodes` don't provide good context on why this is the case, this relationship is formed by the way we use `dependency_graph_` and `missing_nodes_`. Should add some description on how the two member variables work together and this function should name the arguments in the way that reflects this relationship.

Basically it is a two layers (/ groups) graph, here the function is fixing the connection within one layer at a time, while the other argument is the other layer of the whole graph that is not "in action". The arguments must not be arbitrary groups of nodes, they must form a proper graph together.

Reply:
Added more descriptions.